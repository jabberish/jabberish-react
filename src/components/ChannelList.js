import React from 'react';
import PropTypes from 'prop-types';
import ChannelItem from './ChannelItem';

const ChannelList = ({ channels }) => {

  const channelItems = channels.map(channel => {
    return <ChannelItem key={`${channel._id}-${channel.name}`} channel={channel} />;
  });
  return (
    <ul>
      Channel List
      {channelItems}
    </ul>
  );
};

ChannelList.propTypes = {
  channels: PropTypes.array.isRequired
};

export default ChannelList;
