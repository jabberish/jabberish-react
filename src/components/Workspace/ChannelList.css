.ChannelList {
  width: 220px;
  background: #3f0e40;
  list-style: none;
  padding: 0;
}