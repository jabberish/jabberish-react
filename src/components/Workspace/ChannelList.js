import React from 'react';
import PropTypes from 'prop-types';
import ChannelItem from './ChannelItem';

import styles from './ChannelList.css';

const ChannelList = ({ channels, selectChannel }) => {

  const channelItems = channels.map(channel => {
    return (
      <ChannelItem 
        key={`${channel._id}-${channel.name}`} 
        channel={channel} 
        selectChannel={selectChannel}
      />
    );
  });
  
  return (
    <ul className={styles.ChannelList}>
      Channel List
      {channelItems}
    </ul>
  );
};

ChannelList.propTypes = {
  channels: PropTypes.array.isRequired,
  selectChannel: PropTypes.func.isRequired
};

export default ChannelList;
