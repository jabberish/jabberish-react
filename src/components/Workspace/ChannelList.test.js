import React from 'react';
import { shallow } from 'enzyme';
import ChannelList from './ChannelList';

describe('ChannelList component', () => {
  it('renders ChannelList', () => {
    const channels = [];
    // eslint-disable-next-line no-console
    const selectChannel = () => console.log('function');
    const wrapper = shallow(<ChannelList channels={channels} selectChannel={selectChannel} />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
