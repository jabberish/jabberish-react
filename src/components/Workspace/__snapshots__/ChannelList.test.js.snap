// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelList component renders ChannelList 1`] = `
<WithStyles(ForwardRef(Drawer))
  className="makeStyles-drawer-2"
  classes={
    Object {
      "paper": "makeStyles-drawerPaper-3",
    }
  }
  variant="permanent"
>
  <div
    className="makeStyles-toolbar-4"
  />
  Channels
  <WithStyles(ForwardRef(List)) />
</WithStyles(ForwardRef(Drawer))>
`;
